### PR TITLE
UT for match_letters, missing_args and list_rename

### PR DESCRIPTION
**Pull request by teotenn**
This merge request requires review, see comments.

### CODE REVIEW

File: tests/testthat/test-universal_useful.R [hunk 1]
```diff
@@ -81,3 +81,61 @@ test_that("find_by_class works", {
...
+  result_3 <- fn3()
+  expect_length(result_3, 2)
+  expect_equal(result_3, c("a", "b"))
+
+  # Test case 4: Exclude default arguments from missing arguments
+  fn4 <- function(a = 1, b) missing_args()
+  result_4 <- fn4(b = 2)
+  expect_length(result_1, 0)
+  expect_equal(result_1, character(0))
+})
```
Comment (teotenn) [inline tests/testthat/test-universal_useful.R]: Can you check for this set of tests? I'm having some weird problems:

- When I run them individually (i.e., line by line) they all pass.
- When I execute this chunk of code, some of them pass.
- When I run the whole unit test, none of this tests pass.

I'm wondering if it is some issue with the function regarding the environment used by testthat package.

File: tests/testthat/test-universal_useful.R [hunk 2]
```diff
@@ -81,3 +81,61 @@ test_that("find_by_class works", {
...
+
+  # Expected result after renaming
+  expected_result <- list(x = 1, y = 2, z = 3)
+
+  # Apply the function
+  result <- list_rename(data, a = "x", b = "y", c = "z")
+
+  # Check if the result matches the expected result
+  expect_equal(result, expected_result)
+})
```
Comment (teotenn) [inline tests/testthat/test-universal_useful.R]: It seems to me that `list_rename` needs debugging. But maybe I am getting something wrong. Could you have a look at it please?

**Comment by yogat3ch** [comment]
@teotenn Taking a look at this now! Sorry for the delay

File: tests/testthat/test-universal_useful.R (hunk 2 above)
Comment (yogat3ch) [reply tests/testthat/test-universal_useful.R]: @teotenn It looks like the argument names and the arguments were reversed in the test, I switched them and it's working now

File: tests/testthat/test-universal_useful.R (hunk 1 above)
Comment (yogat3ch) [reply tests/testthat/test-universal_useful.R]: @teotenn There was indeed an issue with the `corresponding_call` default value, I think I've fixed it now as the test are running correctly in all three cases now.

**Comment by yogat3ch** [comment]
This looks good, thank you @teotenn

**Pull request merged**